### ISSUE TITLE
[Bug] Pokemon Black/White 2, double battle outside battle subway crash with added pokemon

### ISSUE THREAD

**Issue by blepblops** [Ajarmar/universal-pokemon-randomizer-zx#723]
**Describe the bug**
When randomizing Black/White 2 and adding pokemon to trainers causes a crash during the double battle with Ingo & Emmet outside the Battle Subway.
Both trainers seem to have been assigned 6 pokemon each instead of 6 total, this crashes the game once you've defeated enough pokemon.

**To Reproduce**
1. Load up a legitimate ROM of Pokemon Black 2 / White 2 into the randomizer.
2. Under "Foe Pokemon" enable any randomization and the "Additional pokemon for..." setting (Whichever this battle is assigned to) and crank the number up to maximum
3. Save the ROM & load it into your extremely legitimate hardware
4. Once you've progressed to Nimbassa City engage Ingo & Emmet outside the Battle Subway
5. Notice that the number of pokemon you've defeated does not line up with the number displayed on the UI
6. Defeat approximately 6 pokemon to trigger a crash

**Expected behavior**
I expect double battles are supposed to have at maximum 6 pokemon and not crash.

**System (please complete the following information):**
 - Operating System: Win10
 - Java Version: -
 - Randomizer Version: v4.6.0

**Additional context**
I had also randomized wild pokemon, items and given trainers items.

**Comment by tonguerott** [comment]
I don't know if this will work for sure, but i ran into this same freeze with the double battle on the plasma frigate. i was able to get passed it with a odd fix that might help!

Once you've progressed to Nimbassa City engage Ingo & Emmet outside the Battle Subway
enter the fight with less pokemon in team (i did two on my double fight on frigate and that worked)

I fought the frigate double battle with 2 pokemon in team and I had all the same additional context, but my number of additional mons given to important trainer fight was +2)
If i entered the frigate double battle with a full team, I experienced the crash after the second mon on the enemy team was fainted and replaced. oddly enough, this did also occur when 1 enemy mon and the ally mon fainted and was replaced. Seeing the second scenario got me thinking that it was the amount of mons it was having to keep track of in the battle, so i attempted entering the fight with a partial team and that appears to give less data for the battle to track and allows it to continue without freezing. I will attempt with more and less mons if needed to test for a sweet spot, but two mons worked for me!

TL;DR: I saw same bug with +2 enemy mons, tried with a team of 2 mons so game had less to think about and it let me bypass the freeze point i was stuck in.
